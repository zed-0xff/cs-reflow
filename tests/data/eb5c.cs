	public unsafe static Record_ab57 eb5c6275b8354df1bc9112b5c28f56d9(object object_0wf)
	{
		object obj3 = default(object);
		uint num8 = default(uint);
		int num9 = default(int);
		int tRNS = default(int);
		while (true)
        {
            string text2 = get_str(20179, 32201, 34);
            int num4 = 0;
            while (true)
            {
                if (num4 >= String.get_Length(text2))
                {
                    string text3 = get_str(33061, 44878, 20);
                    int num5 = 0;
                    Record_ab57 obj = new Record_ab57();
                    object obj2 = obj;
                    while (true)
                    {
                        int num6 = 839874299;
                        while (true)
                        {
                            uint num7;
                            int num10;
                            object obj4;
                            string[] obj5;
                            int num11;
                            int num12;
                            object obj6;
                            string[] obj7;
                            int num13;
                            int num14;
                            int num15;
                            int num16;
                            int num17;
                            object obj8;
                            string[] obj9;
                            int num18;
                            int num19;
                            switch ((num7 = (uint)(num6 ^ 0x6EE87003)) % 5) // [3x17]
                            {
                                case 0u:
                                    break;
                                default:
                                    {
                                        object obj10 = object_0wf;
                                        string[] obj11 = new string[1] { Strings_a392.ab50e274fb3141edab7ca8f95fc73273 };
                                        obj3 = String.Split(obj10, obj11, 0);
                                        num10 = ((int)num8 * -1906688326) ^ 0xF11E5F5;
                                        goto IL_022e;
                                    }
                                case 3u:                                         // [1, 2, 3, 4, 5, 6, 7, 8, 9, ...
                                    switch ((num8 = (uint)((~(~((num9 - 1228762352) ^ -1536825443 ^ -1397753263 ^ -1076351360)) * 813229573 * -2052997669) ^ 0x3C5F4E4D)) % 42) // [9, 0, 32, 8, 34, 41, 18, 2, 17, ...
                                    {
                                        case 37u:                                // [13, 34, 55, 76]
                                            break;
                                        case 41u:                                // [6]
                                            goto IL_031a;
                                        default:
                                            goto IL_0339;
                                        case 16u:
                                            goto IL_034c;
                                        case 11u:                                // [20, 41, 62, 83]
                                            goto IL_0356;
                                        case 21u:                                // [9, 30, 51, 72, 93]
                                            goto IL_0369;
                                        case 27u:                                // [8, 29, 50, 71, 92]
                                            goto IL_03c0;
                                        case 4u:
                                            goto IL_03df;
                                        case 39u:                                // [4, 25, 46, 67, 88]
                                            goto IL_03f6;
                                        case 28u:                                // [18, 39, 60, 81]
                                            goto IL_0414;
                                        case 8u:                                 // [4]
                                            goto IL_0457;
                                        case 6u:                                 // [17, 38, 59, 80]
                                            goto IL_0476;
                                        case 10u:                                // [14, 35, 56, 77]
                                            goto IL_049a;
                                        case 38u:                                // [15]
                                            goto IL_04e3;
                                        case 23u:                                // [6, 27, 48, 69, 90]
                                            goto IL_0502;
                                        case 31u:                                // [1, 22, 43, 64, 85]
                                            goto IL_0521;
                                        case 24u:                                // [12, 33, 54, 75]
                                            goto IL_0561;
                                        case 2u:                                 // [8]
                                            goto IL_05a1;
                                        case 22u:                                // [10]
                                            goto IL_05c0;
                                        case 17u:                                // [9]
                                            goto IL_05df;
                                        case 32u:                                // [3]
                                            goto IL_05fe;
                                        case 0u:                                 // [2]
                                            goto IL_061d;
                                        case 26u:                                // [14]
                                            goto IL_063c;
                                        case 1u:                                 // [21, 42, 63, 84]
                                            goto IL_0652;
                                        case 9u:                                 // [1]
                                            goto IL_068c;
                                        case 7u:                                 // [3, 24, 45, 66, 87]
                                            goto IL_06be;
                                        case 5u:                                 // [15, 36, 57, 78]
                                            goto IL_06d3;
                                        case 30u:                                // [12]
                                            goto IL_06f1;
                                        case 13u:                                // [16, 37, 58, 79]
                                            goto IL_0732;
                                        case 18u:                                // [7]
                                            goto IL_0750;
                                        case 33u:
                                            goto IL_076f;
                                        case 36u:                                // [11]
                                            goto IL_0786;
                                        case 25u:                                // [11, 32, 53, 74]
                                            goto IL_07ca;
                                        case 12u:                                // [10, 31, 52, 73]
                                            goto IL_07ed;
                                        case 19u:                                // [16]
                                            goto IL_0833;
                                        case 20u:                                // [19, 40, 61, 82]
                                            goto IL_084e;
                                        case 3u:                                 // [7, 28, 49, 70, 91]
                                            goto IL_088e;
                                        case 35u:
                                            goto IL_08ad;
                                        case 29u:                                // [2, 23, 44, 65, 86]
                                            goto IL_08c4;
                                        case 40u:                                // [13]
                                            goto IL_08d7;
                                        case 14u:                                // [5, 26, 47, 68, 89]
                                            goto IL_08ea;
                                        case 34u:                                // [5]
                                            goto IL_0908;
                                        case 15u:                                // [17]
                                            goto IL_0927;
                                    }
                                    goto default;
                                case 2u:                                         // [10, 32, 54, 76]
                                    goto IL_034c;
                                case 1u:
                                    goto IL_0927;
                                    IL_0927:
                                        return (Record_ab57)obj2;

                                    IL_0908:
                                        ((Record_ab57)obj2).ab1341e71ec24dbcaeab31ed2dbf300c = (string)((object[])obj3)[17];
                                    num10 = ((int)num8 * -575690770) ^ -1870447634;
                                    goto IL_022e;

                                    IL_08ea:
                                        ((Record_ab57)obj2).aa34b551e4724ba78aab8e636ffd32ca = (string)((object[])obj3)[8];
                                    num10 = (int)((num8 * 1340757219) ^ 0x2895FEA4);
                                    goto IL_022e;

                                    IL_08d7:
                                        num10 = ((int)num8 * -480334008) ^ -1419688589;
                                    goto IL_022e;

                                    IL_08c4:
                                        num10 = (int)((num8 * 1236967088) ^ 0x5876264E);
                                    goto IL_022e;

                                    IL_08ad:
                                        ((Record_ab57)obj2).aef7988922e244fc874c4a30beff39b0 = new string[0];
                                    num10 = 37287426;
                                    goto IL_022e;

                                    IL_088e:
                                        ((Record_ab57)obj2).a187aa0a26f441d194c10ab42d0550b0 = (string)((object[])obj3)[10];
                                    num10 = ((int)num8 * -1809642450) ^ -232437250;
                                    goto IL_022e;

                                    IL_084e:
                                        obj4 = ((object[])obj3)[4];
                                    obj5 = new string[1] { Class_efdf.c685d9bd4b0643458370368cdcdc46be };
                                    ((Record_ab57)obj2).aef7988922e244fc874c4a30beff39b0 = String.Split(obj4, obj5, 0);
                                    num10 = (int)(num8 * 1425357082) ^ -1777908704;
                                    goto IL_022e;

                                    IL_0833:
                                        ((Record_ab57)obj2).flag_a961 = true;
                                    num10 = ((int)num8 * -1859360353) ^ -1684071505;
                                    goto IL_022e;

                                    IL_07ed:
                                        object_0wf = String.Substring(object_0wf, String.IndexOf(object_0wf, Strings_a392.e7dbba00b1ec4f6aad59de101cf391aa) + 2);
                                    num10 = ((int)num8 * -1842220977) ^ 0x536E8C54;
                                    goto IL_022e;

                                    IL_07ca:
                                        object_0wf = decrypt_a8f4(object_0wf, string.Empty);
                                    num10 = (int)((num8 * 1283728958) ^ 0x46D3B999);
                                    goto IL_022e;

                                    IL_0786:
                                        if (String.Contains(((object[])obj3)[23], Class_efdf.c685d9bd4b0643458370368cdcdc46be))
                                        {
                                            num11 = -1431533315;
                                            num12 = -1431533315;
                                        }
                                        else
                                        {
                                            num11 = -1875082095;
                                            num12 = -1875082095;
                                        }
                                    num10 = num11 ^ (int)(num8 * 19398123);
                                    goto IL_022e;

                                    IL_03f6:
                                        ((Record_ab57)obj2).af04d80186c046a7aa5cd0a0b4c7fd3e = (string)((object[])obj3)[7];
                                    num10 = ((int)num8 * -634346590) ^ 0x26CF7F59;
                                    goto IL_022e;

                                    IL_03df:
                                        ((Record_ab57)obj2).a54c3b26eb33463fb9b9a4bc1836cb7d = new string[0];
                                    num10 = 756279587;
                                    goto IL_022e;

                                    IL_0339:
                                        num6 = (int)((num7 * 1554655791) ^ 0x2AE17B96);
                                    continue;
                                    IL_076f:
                                        ((Record_ab57)obj2).e771120b0ffa4a18bfaeac420dbc1fa0 = new string[0];
                                    num10 = -846446914;
                                    goto IL_022e;

                                    IL_0750:
                                        ((Record_ab57)obj2).a425a28688d845259efb0dad4303ad19 = (string)((object[])obj3)[19];
                                    num10 = ((int)num8 * -1474346719) ^ 0x997A961;
                                    goto IL_022e;

                                    IL_0732:
                                        ((Record_ab57)obj2).c63b7b445a5349edacb84885c0365709 = (string)((object[])obj3)[2];
                                    num10 = (int)(num8 * 423302001) ^ -845603884;
                                    goto IL_022e;

                                    IL_06f1:
                                        obj6 = ((object[])obj3)[23];
                                    obj7 = new string[1] { Class_efdf.c685d9bd4b0643458370368cdcdc46be };
                                    ((Record_ab57)obj2).a54c3b26eb33463fb9b9a4bc1836cb7d = String.Split(obj6, obj7, 0);
                                    num10 = ((int)num8 * -2102404291) ^ 0x515C4143;
                                    goto IL_022e;

                                    IL_06d3:
                                        ((Record_ab57)obj2).b7079b9dfa9f46589b7f646096517555 = (string)((object[])obj3)[1];
                                    num10 = (int)((num8 * 411889364) ^ 0x7D632722);
                                    goto IL_022e;

                                    IL_06be:
                                        ((Record_ab57)obj2).ac785d30e340485bb6167b24c9574eed = (string)((object[])obj3)[6];
                                    num10 = -732550558;
                                    goto IL_022e;

                                    IL_068c:
                                        ((Record_ab57)obj2).abcd3560a84b498e9ddf821d50965945 = Convert.ToBoolean((string)((object[])obj3)[13]);
                                    num10 = (int)((num8 * 1688120753) ^ 0x2502429A);
                                    goto IL_022e;

                                    IL_0652:
                                        if (!String.Contains(((object[])obj3)[5], Class_efdf.c685d9bd4b0643458370368cdcdc46be))
                                        {
                                            num10 = 1482018352;
                                            num13 = 1482018352;
                                        }
                                        else
                                        {
                                            num10 = 170093716;
                                            num13 = 170093716;
                                        }
                                    goto IL_022e;

                                    IL_03c0:
                                        ((Record_ab57)obj2).a8943f7134114231953e1c4e549f24d3 = (string)((object[])obj3)[11];
                                    num10 = (int)(num8 * 1245029633) ^ -1400460077;
                                    goto IL_022e;

                                    IL_0369:
                                        ((Record_ab57)obj2).a69c07ede66c4d4fb09a60ae59f74ef0 = Convert.ToBoolean((string)((object[])obj3)[12]);
                                    num14 = (int)((uint)num4 / 362426u) - -1073741824;
                                    num15 = afd1f0100a9e48eb9a6c8975d80c2689;
                                    if (num14 == (int)((0 - (uint)num15 / 3185410u) / 64))
                                    {
                                        goto end_IL_01b2;
                                    }
                                    num10 = (int)((num8 * 1785997207) ^ 0x6E704129);
                                    goto IL_022e;

                                    IL_063c:
                                        ((Record_ab57)obj2).f6727ce02cd8425189354b236d6b8340 = (string)((object[])obj3)[24];
                                    num10 = 550581587;
                                    goto IL_022e;

                                    IL_061d:
                                        ((Record_ab57)obj2).a69db68d95324c51b71feeafde57b763 = (string)((object[])obj3)[14];
                                    num10 = (int)(num8 * 439328159) ^ -1305451641;
                                    goto IL_022e;

                                    IL_05fe:
                                        ((Record_ab57)obj2).a77a37dd25af48738c21a909fc9b7842 = (string)((object[])obj3)[15];
                                    num10 = ((int)num8 * -1835375722) ^ -960561697;
                                    goto IL_022e;

                                    IL_05df:
                                        ((Record_ab57)obj2).ecdd41ece05e4488892c937b5985f01a = (string)((object[])obj3)[21];
                                    num10 = (int)((num8 * 1823611540) ^ 0x59326C41);
                                    goto IL_022e;

                                    IL_05c0:
                                        ((Record_ab57)obj2).a9e740d690154b578e5242847f15efff = (string)((object[])obj3)[22];
                                    num10 = (int)(num8 * 338281772) ^ -1788910821;
                                    goto IL_022e;

                                    IL_05a1:
                                        ((Record_ab57)obj2).f34fababc74b465cb090f85cea5106df = (string)((object[])obj3)[20];
                                    num10 = ((int)num8 * -578543901) ^ -643547062;
                                    goto IL_022e;

                                    IL_0561:
                                        if (String.Contains(object_0wf, Strings_a392.ab50e274fb3141edab7ca8f95fc73273))
                                        {
                                            num16 = 635804220;
                                            num17 = 635804220;
                                        }
                                        else
                                        {
                                            num16 = -185364284;
                                            num17 = -185364284;
                                        }
                                    num10 = num16 ^ (int)(num8 * 245212867);
                                    goto IL_022e;

                                    IL_0356:
                                        num10 = ((int)num8 * -1082901484) ^ 0x65A66076;
                                    goto IL_022e;

                                    IL_034c:
                                        num10 = 1278826133;
                                    goto IL_022e;

                                    IL_022e:
                                        num9 = num10;
                                    num6 = 145409447;
                                    continue;

                                    IL_0521:
                                        obj8 = ((object[])obj3)[5];
                                    obj9 = new string[1] { Class_efdf.c685d9bd4b0643458370368cdcdc46be };
                                    ((Record_ab57)obj2).e771120b0ffa4a18bfaeac420dbc1fa0 = String.Split(obj8, obj9, 0);
                                    num10 = (int)((num8 * 1622542490) ^ 0x2F2BFDBE);
                                    goto IL_022e;

                                    IL_0502:
                                        ((Record_ab57)obj2).a29ea6e4ce884adb81efc5f82cd9c145 = (string)((object[])obj3)[9];
                                    num10 = ((int)num8 * -620582044) ^ -1268718394;
                                    goto IL_022e;

                                    IL_04e3:
                                        ((Record_ab57)obj2).a86721fc31bc49d4a040f2e7aa5418f8 = (string)((object[])obj3)[25];
                                    num10 = ((int)num8 * -394500051) ^ -388323700;
                                    goto IL_022e;

                                    IL_049a:
                                        ((Record_ab57)obj2).ab2645063e1e46d88600e382d1748ce0 = Text.Encoding.get_UTF8().GetString(Convert.FromBase64String((string)((object[])obj3)[0]));
                                    num10 = (int)((num8 * 1098850859) ^ 0x186E6A3C);
                                    goto IL_022e;

                                    IL_0476:
                                        ((Record_ab57)obj2).a09b41ccc00e4a40a4a434f6007b3f89 = (string)((object[])obj3)[3];
                                    tRNS = Structs_e0d5.tRNS;
                                    num10 = (int)((num8 * 2044808620) ^ 0x4DF9B037);
                                    goto IL_022e;

                                    IL_0457:
                                        ((Record_ab57)obj2).a508b9a874b34ddea2cd2561652cf051 = (string)((object[])obj3)[16];
                                    num10 = (int)((num8 * 1550262346) ^ 0x6969162F);
                                    goto IL_022e;

                                    IL_0414:
                                        if (String.Contains(((object[])obj3)[4], Class_efdf.c685d9bd4b0643458370368cdcdc46be))
                                        {
                                            num18 = -1051179905;
                                            num19 = -1051179905;
                                        }
                                        else
                                        {
                                            num18 = 2027452966;
                                            num19 = 2027452966;
                                        }
                                    num10 = num18 ^ (int)(num8 * 1490210903);
                                    goto IL_022e;

                                    IL_031a:
                                        ((Record_ab57)obj2).a8c2197d59e242fabb26eb9459883b8a = (string)((object[])obj3)[18];
                                    num10 = ((int)num8 * -622770849) ^ 0x48AEFFBE;
                                    goto IL_022e;
                            }
                            break;
                        }
                        continue;
                        end_IL_01b2:
                            break;
                    }
                }
                int iend = Structs_e0d5.IEND;
                if ((-4075 | iend) - 14642 == iend * (0xF15 ^ (tRNS + 3 * tRNS)))
                {
                    break;
                }
                num4++;
            }
        }
	}
