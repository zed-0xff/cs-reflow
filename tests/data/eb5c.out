	public unsafe static Record_ab57 eb5c6275b8354df1bc9112b5c28f56d9(object object_0wf)
	{
	    object obj3 = default(object);
	    lbl_6:
	        string text2 = get_str(20179, 32201, 34);
	    int num4 = 0;
	    while (true)
	    {
	        if (num4 >= String.get_Length(text2))
	        {
	            string text3 = get_str(33061, 44878, 20);
	            Record_ab57 obj = new Record_ab57();
	            object obj2 = obj;
	            object_0wf = String.Substring(object_0wf, String.IndexOf(object_0wf, Strings_a392.e7dbba00b1ec4f6aad59de101cf391aa) + 2);
	            object_0wf = decrypt_a8f4(object_0wf, string.Empty);
	            if (String.Contains(object_0wf, Strings_a392.ab50e274fb3141edab7ca8f95fc73273))
	            {
	                object obj10 = object_0wf;
	                string[] obj11 = new string[1]
	                {
	                    Strings_a392.ab50e274fb3141edab7ca8f95fc73273
	                };
	                obj3 = String.Split(obj10, obj11, 0);
	                ((Record_ab57)obj2).ab2645063e1e46d88600e382d1748ce0 = Text.Encoding.get_UTF8().GetString(Convert.FromBase64String((string)((object[])obj3)[0]));
	                ((Record_ab57)obj2).b7079b9dfa9f46589b7f646096517555 = (string)((object[])obj3)[1];
	                ((Record_ab57)obj2).c63b7b445a5349edacb84885c0365709 = (string)((object[])obj3)[2];
	                ((Record_ab57)obj2).a09b41ccc00e4a40a4a434f6007b3f89 = (string)((object[])obj3)[3];
	                if (String.Contains(((object[])obj3)[4], Class_efdf.c685d9bd4b0643458370368cdcdc46be))
	                {
	                    object obj4;
	                    string[] obj5;
	                    obj4 = ((object[])obj3)[4];
	                    obj5 = new string[1]
	                    {
	                        Class_efdf.c685d9bd4b0643458370368cdcdc46be
	                    };
	                    ((Record_ab57)obj2).aef7988922e244fc874c4a30beff39b0 = String.Split(obj4, obj5, 0);
	                }
	                else
	                {
	                    ((Record_ab57)obj2).aef7988922e244fc874c4a30beff39b0 = new string[0];
	                }
	
	                if (!String.Contains(((object[])obj3)[5], Class_efdf.c685d9bd4b0643458370368cdcdc46be))
	                {
	                    ((Record_ab57)obj2).e771120b0ffa4a18bfaeac420dbc1fa0 = new string[0];
	                }
	                else
	                {
	                    object obj8;
	                    string[] obj9;
	                    obj8 = ((object[])obj3)[5];
	                    obj9 = new string[1]
	                    {
	                        Class_efdf.c685d9bd4b0643458370368cdcdc46be
	                    };
	                    ((Record_ab57)obj2).e771120b0ffa4a18bfaeac420dbc1fa0 = String.Split(obj8, obj9, 0);
	                }
	
	                ((Record_ab57)obj2).ac785d30e340485bb6167b24c9574eed = (string)((object[])obj3)[6];
	                ((Record_ab57)obj2).af04d80186c046a7aa5cd0a0b4c7fd3e = (string)((object[])obj3)[7];
	                ((Record_ab57)obj2).aa34b551e4724ba78aab8e636ffd32ca = (string)((object[])obj3)[8];
	                ((Record_ab57)obj2).a29ea6e4ce884adb81efc5f82cd9c145 = (string)((object[])obj3)[9];
	                ((Record_ab57)obj2).a187aa0a26f441d194c10ab42d0550b0 = (string)((object[])obj3)[10];
	                ((Record_ab57)obj2).a8943f7134114231953e1c4e549f24d3 = (string)((object[])obj3)[11];
	                ((Record_ab57)obj2).a69c07ede66c4d4fb09a60ae59f74ef0 = Convert.ToBoolean((string)((object[])obj3)[12]);
	                ((Record_ab57)obj2).abcd3560a84b498e9ddf821d50965945 = Convert.ToBoolean((string)((object[])obj3)[13]);
	                ((Record_ab57)obj2).a69db68d95324c51b71feeafde57b763 = (string)((object[])obj3)[14];
	                ((Record_ab57)obj2).a77a37dd25af48738c21a909fc9b7842 = (string)((object[])obj3)[15];
	                ((Record_ab57)obj2).a508b9a874b34ddea2cd2561652cf051 = (string)((object[])obj3)[16];
	                ((Record_ab57)obj2).ab1341e71ec24dbcaeab31ed2dbf300c = (string)((object[])obj3)[17];
	                ((Record_ab57)obj2).a8c2197d59e242fabb26eb9459883b8a = (string)((object[])obj3)[18];
	                ((Record_ab57)obj2).a425a28688d845259efb0dad4303ad19 = (string)((object[])obj3)[19];
	                ((Record_ab57)obj2).f34fababc74b465cb090f85cea5106df = (string)((object[])obj3)[20];
	                ((Record_ab57)obj2).ecdd41ece05e4488892c937b5985f01a = (string)((object[])obj3)[21];
	                ((Record_ab57)obj2).a9e740d690154b578e5242847f15efff = (string)((object[])obj3)[22];
	                if (String.Contains(((object[])obj3)[23], Class_efdf.c685d9bd4b0643458370368cdcdc46be))
	                {
	                    object obj6;
	                    string[] obj7;
	                    obj6 = ((object[])obj3)[23];
	                    obj7 = new string[1]
	                    {
	                        Class_efdf.c685d9bd4b0643458370368cdcdc46be
	                    };
	                    ((Record_ab57)obj2).a54c3b26eb33463fb9b9a4bc1836cb7d = String.Split(obj6, obj7, 0);
	                }
	                else
	                {
	                    ((Record_ab57)obj2).a54c3b26eb33463fb9b9a4bc1836cb7d = new string[0];
	                }
	
	                ((Record_ab57)obj2).f6727ce02cd8425189354b236d6b8340 = (string)((object[])obj3)[24];
	                ((Record_ab57)obj2).a86721fc31bc49d4a040f2e7aa5418f8 = (string)((object[])obj3)[25];
	                ((Record_ab57)obj2).flag_a961 = true;
	            }
	
	            return (Record_ab57)obj2;
	        }
	        else
	        {
	            num4++;
	        }
	    }
	
	    goto lbl_6;
	}

